MuvrKit/MKWatchConnectivity.swift

```diff
@@ -193,28 +244,37 @@ public final class MKConnectivity : NSObject, WCSessionDelegate {
             let from = props.accelerometerStart ?? session.start
             let to = props.end ?? NSDate()
             
-            // if this session is meant to end, we require all sensor data to be available.
-            // there is a neater way, but this saves memory
-            guard let sensorData = getSamples(from: from, to: to, requireAll: props.end != nil, demo: session.demo) else {
-                NSLog("Not enough sensor data in \(from) - \(to)")
+            guard let sensorData = getSamples(from: from, to: to, demo: session.demo) else {
+                NSLog("No sensor data in \(from) - \(to)")
                 return
             }
 
-            // set the expected range of samples on the next call
-            let updatedProps = props.with(accelerometerStart: from.dateByAddingTimeInterval(sensorData.duration), recorded: sensorData.rowCount)
+            // update the number of recorded samples
+            let readFromDate = NSDate(timeIntervalSince1970: sensorData.end)
+            let updatedProps = props.with(accelerometerEnd: readFromDate)
             self.sessions[session] = updatedProps
             
             // transfer what we have so far
-            transferSensorDataBatch(sensorData, session: session, props: props) {
-                // update the session with incremented sent counter
-                if props.end == nil {
-                    self.sessions[session] = updatedProps.with(sent: sensorData.rowCount)
-                } else {
-                    self.sessions.removeValueForKey(session)
-                    // we're done with this session, we can move on to the next one
+            transferSensorDataBatch(sensorData, session: session, props: updatedProps) {
```

Comment:
I think this might be "des Pudels Kern": this is the only place where sessions could magically be inserted back into the map.